Content.Server/Administration/Systems/BwoinkSystem.cs

```diff
@@ -180,6 +181,19 @@ private async void OnPlayerStatusChanged(object? sender, SessionStatusEventArgs
                     _activeConversations.Remove(e.Session.UserId);
                     return;
                 }
+
+                // check if the username has been banned
+                var lookup = await _playerLocator.LookupIdAsync(e.Session.UserId);
+
+                if (lookup != null) {
+                    var usernameBanReason = _usernameRules.IsUsernameBannedAsync(lookup.Username);
```

Comment:
`await`?  And for the `Loc.GetString`, shouldn't you pass `usernameBanReason.Message` instead of the whole struct, or does fluent somehow handle that.